src/GuzzleRequestAdapter.php

```diff
@@ -281,152 +356,282 @@ public function getBaseUrl(): string
-                $this->getPsrRequestFromRequestInformation($authenticatedRequest)
-        );
+        $currentContext = Context::getCurrent();
+        $thisSpan = $this->tracer->spanBuilder('convertToNative');
+        if ($span !== null) {
+            $thisSpan = $thisSpan->setParent($span->storeInContext($currentContext));
+        }
+        $thisSpan = $thisSpan->startSpan();
+        try {
+            $result = $this->authenticationProvider->authenticateRequest($requestInformation)->then(
+                fn(RequestInformation $authenticatedRequest): RequestInterface => $this->getPsrRequestFromRequestInformation($authenticatedRequest, $span)
+            );
+        } finally {
+            $thisSpan->end();
+            if ($span !== null) $span->end();
+        }
+        return $result;
     }
 
     /**
      * Gets the root parse node using the parseNodeFactory based on the Content-Type
      *
      * @param ResponseInterface $response
+     * @param SpanInterface $span
      * @return ParseNode
      */
-    private function getRootParseNode(ResponseInterface $response): ParseNode
+    private function getRootParseNode(ResponseInterface $response, SpanInterface $span): ParseNode
     {
-        if (!$response->hasHeader(RequestInformation::$contentTypeHeader)) {
-            throw new RuntimeException("No response content type header for deserialization");
+        $serializationSpan = $this->tracer->spanBuilder('ParseNode::getObjectValue')
+            ->addLink($span->getContext())
+            ->startSpan();
+        $scope = $serializationSpan->activate();
+        try {
+            if (!$response->hasHeader(RequestInformation::$contentTypeHeader)) {
+                throw new RuntimeException("No response content type header for deserialization");
+            }
+            $contentType = explode(';', $response->getHeaderLine(RequestInformation::$contentTypeHeader));
+
+            $result = $this->parseNodeFactory->getRootParseNode($contentType[0], $response->getBody());
+            $serializationSpan->setStatus(StatusCode::STATUS_OK, 'deserialize_success');
+            return $result;
+        } finally {
+            $scope->detach();
+            $serializationSpan->end();
         }
-        $contentType = explode(';', $response->getHeaderLine(RequestInformation::$contentTypeHeader));
-        return $this->parseNodeFactory->getRootParseNode($contentType[0], $response->getBody());
     }
 
     /**
      * Authenticates and executes the request
      *
      * @param RequestInformation $requestInformation
      * @param string $claims additional claims to request if CAE fails
+     * @param SpanInterface|null $span
      * @return Promise
      */
-    private function getHttpResponseMessage(RequestInformation $requestInformation, string $claims = ""): Promise
+    private function getHttpResponseMessage(RequestInformation $requestInformation, string $claims = '', ?SpanInterface $span = null): Promise
```

Comment:
Some attributes are missing here. Check this file for the required and optional attributes https://github.com/microsoftgraph/msgraph-sdk-design/blob/9974f7ed0baa99aeef92d19098bd239696c697db/Observability.md#what-to-capture---tracing